graphs/models.py

```diff
@@ -310,6 +310,16 @@ class Edge(Base):
             ForeignKeyConstraint([user_id, graph_id, tail_node_id], [Node.user_id, Node.graph_id, Node.node_id], ondelete="CASCADE", onupdate="CASCADE"), {})
     #no relationship specified
 
+class share_graph_event(Base):
+    __tablename__ = 'share_graph_event'
+    id = Column(Integer, autoincrement=True, primary_key=True)
+    graph_id = Column(String, ForeignKey('graph.graph_id', ondelete="CASCADE", onupdate="CASCADE"), nullable=False)
+    owner_id = Column(String, ForeignKey('group.owner_id', ondelete="CASCADE", onupdate="CASCADE"), nullable=False)
+    group_id = Column(String, ForeignKey('group.group_id', ondelete="CASCADE", onupdate="CASCADE"), nullable=False)
+    member_id = Column(String, ForeignKey('user.user_id', ondelete="CASCADE", onupdate="CASCADE"), nullable=False)
+    share_time = Column(TIMESTAMP, nullable = False)
+    is_active = Column(Boolean)
```

Comment:
It should be enforced in the db, especially we *can *enforce it at the
schema level.


On Sun, Jul 24, 2016 at 10:28 AM, Mridul Seth <notifications@github.com>
wrote:

> In graphs/models.py
> <https://github.com/Murali-group/GraphSpace/pull/191#discussion_r71989505>
> :
>
> > @@ -310,6 +310,16 @@ class Edge(Base):
> >              ForeignKeyConstraint([user_id, graph_id, tail_node_id], [Node.user_id, Node.graph_id, Node.node_id], ondelete="CASCADE", onupdate="CASCADE"), {})
> >      #no relationship specified
> >
> > +class share_graph_event(Base):
> > +    __tablename__ = 'share_graph_event'
> > +    id = Column(Integer, autoincrement=True, primary_key=True)
> > +    graph_id = Column(String, ForeignKey('graph.graph_id', ondelete="CASCADE", onupdate="CASCADE"), nullable=False)
> > +    owner_id = Column(String, ForeignKey('group.owner_id', ondelete="CASCADE", onupdate="CASCADE"), nullable=False)
> > +    group_id = Column(String, ForeignKey('group.group_id', ondelete="CASCADE", onupdate="CASCADE"), nullable=False)
> > +    member_id = Column(String, ForeignKey('user.user_id', ondelete="CASCADE", onupdate="CASCADE"), nullable=False)
> > +    share_time = Column(TIMESTAMP, nullable = False)
> > +    is_active = Column(Boolean)
>
> Are you not adding the unique constraint because you are thinking of
> someone unsharing a graph and then sharing it again?
>
> A user can unshare the graph and as discussed in the meeting with Ted we
> can delete the share event.
> But I think the unique constraint part will be a part of the logic
> implemented in the db.py file.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Murali-group/GraphSpace/pull/191/files/c79c1bd7ef35f3349c121b770faa993665cf6d34#r71989505>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGkWUG8U3i4hJ6l_OPDOGCeJCw2w1TSHks5qY3ZwgaJpZM4JS_ZW>
> .
>